src/benchmark/runner.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  File,
+  startTests,
+  Suite,
+  Task,
+  VitestRunner,
+  VitestRunnerConfig,
+  VitestRunnerImportSource,
+} from "@vitest/runner";
+import {Benchmark, BenchmarkOpts, BenchmarkResults} from "../types.js";
+import {BenchmarkReporter} from "./reporter.js";
+import {store} from "./globalState.js";
+
+export class BenchmarkRunner implements VitestRunner {
+  readonly config: VitestRunnerConfig;
+  readonly reporter: BenchmarkReporter;
+  readonly prevBench: Benchmark | null;
+  readonly benchmarkOpts: BenchmarkOpts;
+
+  constructor({prevBench, benchmarkOpts}: {prevBench: Benchmark | null; benchmarkOpts: BenchmarkOpts}) {
+    this.config = {
+      root: "",
+      sequence: {seed: 1234, hooks: "list", setupFiles: "list"},
+      passWithNoTests: false,
+      maxConcurrency: 1,
+      hookTimeout: 10_0000,
+      testTimeout: 10_0000,
+      setupFiles: [],
+      retry: 0,
+    };
+    this.prevBench = prevBench;
+    this.benchmarkOpts = benchmarkOpts;
+    this.reporter = new BenchmarkReporter({prevBench, benchmarkOpts});
+  }
+
+  onBeforeRunSuite(suite: Suite): void {
+    this.reporter.onSuiteStarted(suite);
+  }
+
+  onAfterRunSuite(suite: Suite): void {
+    this.reporter.onSuiteFinished(suite);
+  }
+
+  onBeforeRunTask(task: Task): void {
+    this.reporter.onTestStarted(task);
+  }
+
+  onAfterRunTask(task: Task): void {
+    this.reporter.onTestFinished(task);
+  }
+
+  onAfterRunFiles(files: File[]): void {
+    this.reporter.onComplete(files);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async importFile(filepath: string, _source: VitestRunnerImportSource): Promise<void> {
+    // TODO: Implement file caching mechanism later
```

Comment:
Why not sue the `VitestRunnerImportSource`?  Docs say that is exactly what its for and guessing that also implements the caching mechanism mentioned in the comment.

Reply:
Importing is not in the scope of the `@vitest/runner`. In the `vitest` they have implemented it for different environment differently e.g. node or browser. 

I don't want to have our benchmark tool been dependent on the `vitest`.